apps/cowswap-frontend/src/common/pure/TradeDetailsAccordion/index.tsx

```diff
@@ -1,135 +1,57 @@
 
 import CarretIcon from '@cowprotocol/assets/cow-swap/carret-down.svg'
-import { FiatAmount, TokenAmount, UI } from '@cowprotocol/ui'
+import { FiatAmount, TokenAmount } from '@cowprotocol/ui'
 import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
 
 import SVG from 'react-inlinesvg'
-import styled from 'styled-components/macro'
 
-const Wrapper = styled.div<{ isOpen: boolean }>`
-  display: flex;
-  flex-flow: row wrap;
-  align-items: center;
-  width: 100%;
-  border: 1px solid var(${UI.COLOR_PAPER_DARKER});
-  border-radius: 16px;
-  padding: 10px;
-  height: auto;
-  font-size: 13px;
-  font-weight: var(${UI.FONT_WEIGHT_MEDIUM});
-`
-
-const Details = styled.div`
-  display: flex;
-  flex-flow: row wrap;
-  align-items: center;
-  width: 100%;
-  margin: 6px 0 0;
-`
-
-const Summary = styled.div`
-  display: grid;
-  grid-template-columns: auto 1fr;
-  justify-content: space-between;
-  align-items: center;
-  width: 100%;
-  gap: 8px;
-  font-size: inherit;
-  font-weight: inherit;
-
-  span {
-    font-size: inherit;
-    font-weight: inherit;
-    white-space: nowrap;
-  }
-`
-
-const SummaryClickable = styled.div<{ isOpen: boolean }>`
-  cursor: pointer;
-  display: flex;
-  align-items: center;
-  justify-content: flex-end;
-  gap: 7px;
-  opacity: ${({ isOpen }) => (isOpen ? 1 : 0.7)};
-  transition: opacity ${UI.ANIMATION_DURATION_SLOW} ease-in-out;
-  outline: none;
-  font-size: inherit;
-  font-weight: inherit;
-
-  &:hover {
-    opacity: 1;
-  }
-
-  // If it's the only child, make it take all the space
-  &:only-child {
-    grid-column: 1 / -1;
-  }
-`
-
-const ToggleIcon = styled.div<{ isOpen: boolean }>`
-  --size: var(${UI.ICON_SIZE_SMALL});
-  transform: ${({ isOpen }) => (isOpen ? 'rotate(180deg)' : 'rotate(0deg)')};
-  transition: transform var(${UI.ANIMATION_DURATION_SLOW}) ease-in-out;
-  display: flex;
-  align-items: center;
-  justify-content: center;
-  width: var(--size);
-  height: var(--size);
-
-  > svg {
-    --size: var(${UI.ICON_SIZE_TINY});
-    width: var(--size);
-    height: var(--size);
-    object-fit: contain;
-
-    path {
-      fill: var(${UI.COLOR_TEXT});
-    }
-  }
-`
+import { Wrapper, Summary, SummaryClickable, ToggleIcon, Details } from './styled'
 
 interface TradeDetailsAccordionProps {
   rateInfo: React.ReactNode
   feeTotalAmount: CurrencyAmount<Currency> | null
   feeUsdTotalAmount: CurrencyAmount<Currency> | null
   children?: React.ReactNode
-  open?: boolean
+  open: boolean
+  onToggle: () => void
 }
 
 export const TradeDetailsAccordion = ({
   rateInfo,
   feeTotalAmount,
   feeUsdTotalAmount,
   children,
-  open = false,
+  open,
+  onToggle,
 }: TradeDetailsAccordionProps) => {
-  const [isOpen, setIsOpen] = useState(open)
-
-  const toggleAccordion = () => {
-    setIsOpen(!isOpen)
+  const handleToggle = () => {
+    if (typeof onToggle === 'function') {
+      onToggle()
+    } else {
+      console.error('onToggle is not a function:', onToggle)
+    }
```

Comment:
Why do you expect it not to be a function?
According to the type, it should always be one.

You could also:

```suggestion
    onToggle?.()
```

Reply:
Thanks for the suggestion! I think this was a debugging feature I didn't pay attention to anymore.